### ISSUE TITLE
Page endpoint without a GET handler causes client-side navigation to hang

### ISSUE THREAD

**Issue by Rich-Harris** [sveltejs/kit#4253]
### Describe the bug

If a page has an endpoint without a `get` handler, client-side navigation hands indefinitely

### Reproduction

Go to https://stackblitz.com/edit/sveltejs-kit-template-default-v3d3wq?file=src%2Froutes%2Fabout.js&terminal=dev, click the `ABOUT` link in the nav

### System Info

```shell
System:
    OS: macOS 12.0.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 169.98 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.13.1/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
  Browsers:
    Chrome: 98.0.4758.109
    Firefox: 97.0.1
    Safari: 15.1
  npmPackages:
    svelte: ^3.43.0 => 3.44.2
```

### Severity

annoyance

**Comment by am283721** [comment]
Interestingly, I see the issue in the repro but I can't seem to reproduce it locally on any browser. It might be worth noting that it fails to build the project without setting `prerender` to false on the about page (due to having an endpoint with mutative methods).

An error in the stackblitz repro eventually did display (I think triggered by the anchor prefetch):

<img width="769" alt="Screen Shot 2022-03-07 at 10 49 23 AM" src="https://user-images.githubusercontent.com/27497493/157072976-0495b922-6105-4bd7-a4b7-9e77b77e34f3.png">


The error in `start.js` is from trying to fetch from `__data.json`:

```js
if (is_shadow_page) {
  const res = await fetch(
	  `${url.pathname}${url.pathname.endsWith('/') ? '' : '/'}__data.json${url.search}`,
	  {
		  headers: {
			  'x-sveltekit-load': /** @type {string} */ (shadow_key)
		  }
	  }
  );
....
```

**Comment by lsl** [comment]
Any chance this is related to [this change](https://github.com/sveltejs/kit/pull/4203/files#diff-6bc786c5d82299b5eed36bb073cd7e3c8347af2951019da2b5dff1d5e3bad611R626)?

204 is now being used to mean both fallthrough and redirect. Routes are loading shadow pages for less specific routes (e.g a wildcard route when the first route responds with nothing (or in my case, a redirect).

**Comment by WholeStackDev** [comment]
@lsl I confirmed that the issue does not occur when running @sveltejs/kit 1.0.0-next.291 but does occur when running 1.0.0-next.292. The change you mentioned, #4203, did go in on release 292, so that sounds like the culprit. I don't know enough to be able to say why it broke it, but I can at least narrow stuff down!

Here are the repros showing it working on 291 and breaking on 292:

291 = https://stackblitz.com/edit/sveltejs-kit-template-default-yyw5ho
292 = https://stackblitz.com/edit/sveltejs-kit-template-default-dbtggu

**Comment by tadmccorkle** [comment]
Is this still an issue with SvelteKit or just something off with StackBlitz? StackBlitz is the only place I've been able to reproduce the issue. I can still reproduce it there as of SvelteKit 1.0.0-next.310, although it's a little finicky - it won't hang if I quickly hit the `About` link after HMR, but it starts to hang a few seconds later.

I've tried to reproduce locally from scratch and by copying my StackBlitz project source files, but it never hangs when navigating to a page endpoint w/o a GET handler.

Also, there seems to be a test that currently passes for this exact case:
```
cd packages/kit/test/apps/basics/
npx playwright test -g 'Endpoint without GET'
```

**Comment by tcc-sejohnson** [comment]
I concur that this seems to be an issue with StackBlitz -- I tried it on both Chrome and Firefox on Windows and Ubuntu, and couldn't replicate it in any of those, though the issue still persists on next-347 on StackBlitz.

**Comment by andyliuhaowen** [comment]
Same. Cannot repro locally. Would suggest closing this as a non-SvelteKit issue.

**Comment by Rich-Harris** [comment]
Locally, `about/__data.json` is a 204 (what?) response with the following contents:

```
GET method not allowed
```

That shouldn't happen. Browsers seem to take it in their stride, but presumably the service worker that handles requests in StackBlitz doesn't like it. Even though it 'works' in browsers, StackBlitz is exposing a legitimate bug, so I think this needs to stay open.

**Comment by dummdidumm** [comment]
This was fixed as part of #5778 since __data.json is now always present because it queries not only the page endpoint, but everything leading up to it.
